pkg/importer/dataStream_test.go

```diff
@@ -79,239 +172,99 @@ func TestNewDataStream(t *testing.T) {
+			if err := ds.Close(); (err != nil) != tt.wantErr {
 				t.Errorf("dataStream.Close() error = %v, wantErr %v", err, tt.wantErr)
 			}
 		})
 	}
 }
 
 func Test_dataStream_dataStreamSelector(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
-	tests := []struct {
-		name    string
-		fields  fields
-		wantErr bool
-	}{
-	// TODO: Add test cases.
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			d := &dataStream{
-				Url:         tt.fields.Url,
-				Readers:     tt.fields.Readers,
-				buf:         tt.fields.buf,
-				qemu:        tt.fields.qemu,
-				Size:        tt.fields.Size,
-				accessKeyId: tt.fields.accessKeyId,
-				secretKey:   tt.fields.secretKey,
-			}
-			if err := d.dataStreamSelector(); (err != nil) != tt.wantErr {
-				t.Errorf("dataStream.dataStreamSelector() error = %v, wantErr %v", err, tt.wantErr)
-			}
-		})
-	}
-}
-
-func Test_dataStream_s3(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
-	tests := []struct {
-		name    string
-		fields  fields
-		want    *Reader
-		wantErr bool
-	}{
-	// TODO: Add test cases.
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			d := dataStream{
-				Url:         tt.fields.Url,
-				Readers:     tt.fields.Readers,
-				buf:         tt.fields.buf,
-				qemu:        tt.fields.qemu,
-				Size:        tt.fields.Size,
-				accessKeyId: tt.fields.accessKeyId,
-				secretKey:   tt.fields.secretKey,
-			}
-			got, err := d.s3()
-			if (err != nil) != tt.wantErr {
-				t.Errorf("dataStream.s3() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("dataStream.s3() = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
+	imageDir, _ := filepath.Abs("../../test/images")
+	localImageBase := filepath.Join("file://", imageDir)
 
-func Test_dataStream_http(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
 	tests := []struct {
-		name    string
-		fields  fields
-		want    *Reader
-		wantErr bool
+		name     string
+		testFile string
+		ep       string
+		wantErr  bool
 	}{
-	// TODO: Add test cases.
+		{
+			name:     "success building selector for file",
+			testFile: "sample.tar.xz",
+			ep:       localImageBase,
+			wantErr:  false,
+		},
+		{
+			name:     "fail trying to build selector for invalid http endpoint",
+			testFile: "",
+			ep:       "http://www.google.com",
+			wantErr:  true,
+		},
+		{
+			name:     "fail trying to build invalid selector",
+			testFile: "sample.tar.xz",
+			ep:       "fake://somefakefile",
```

Comment:
this will be an endpoint parsing err...

Reply:
well, does it matter for this test? I think that is what I'm going for, the error I get is what I expected? The error doesn't change if I add an extra `/` btw...

```
    	dataStream_test.go:244: dataStream.dataStreamSelector() error = invalid url scheme: "fake", wantErr false
```